CMakeLists.txt

```diff
@@ -61,12 +64,6 @@ endif()
 
 include_directories ("${PROJECT_SOURCE_DIR}")
 
-if(UNIX)
-  set(LIBHEIF_LIBRARY_NAME heif)
```

Comment:
This is necessary so the resulting file on Linux is named `libheif.so`. Without it, the name is `liblibheif.so`:
https://travis-ci.org/strukturag/libheif/jobs/599715352#L664
```
[ 43%] Linking CXX shared library liblibheif.so
```

Reply:
This is due to the internal CMake variable `CMAKE_SHARED_LIBRARY_PREFIX` (whose default value for UNIX environments is `lib`: 

https://cmake.org/cmake/help/latest/variable/CMAKE_SHARED_LIBRARY_PREFIX.html

Would it be fine for you to change the clear the value of that variable on UNIX so that we end up having `libheif.so` on UNIX and `libheif.lib` on Windows ? 

My objective was to remove `LIBHEIF_LIBRARY_NAME` because we already have one variable (`PROJECT_NAME`) containing the information we need.

Reply:
I'm fine with whatever works :smile: However if you clear `CMAKE_SHARED_LIBRARY_PREFIX`, make sure the `libpixbufloader-heif.so` from folder `gdk-pixbuf` still has the `lib` prefix.

Reply:
Ha, I did not take into account that target. I think I will revert then this change and come back to the original implementation.

Reply:
I take back what I just said. If we have different names for the target for different platforms, it is going to be a mess to consume **libheif** from other multi-platform projects. 

I will think about this and come with a proposal.

Reply:
I just made a change related to that. I renamed the `project()` to `heif` so that it will produce:

- Unix: libheif.so
- Windows: heif.lib & heif.dll

Then I hardcoded the include directory to `libheif` so that consumers of the library does not need to change anything to find the header files. It will only change for windows consumers, which will need to update the library name from `libheif` to `heif`. 

From my point of view, it is legit to make this change because anyways, any CMake consumer of **libheif** should now use the CMake Config file to find the dependency, and that config file is in charge of indicating the library name and include directories. Note that it is also the default behaviour in CMake to have libXXX.so for UNIX and XXX.lib on Windows. Experiments trying to modify the default behavior end up in additional code for the maintained project, and additional difficulties for the consumers of the library.

Since this is a "breaking change" for windows users, I propose to create a new release of the library if this PR gets finally merged.